module/VuFind/src/VuFind/Search/Base/Options.php

```diff
@@ -969,7 +969,19 @@ public function getRecommendationSettings($handler = null)
     public function getSearchClassId()
     {
         // Parse identifier out of class name of format VuFind\Search\[id]\Options:
-        $class = explode('\\', get_class($this));
+        $className = get_class($this);
+        $class = explode('\\', $className);
+
+        // Special case: if there's only one part, we may be testing with a mock
+        // object; if so, that's okay, but anything else is unexpected.
+        if (count($class) < 3) {
+            $parts = explode('_', $class[0]);
```

Comment:
Since we don't need the parts, I think it would be cleaner to just check with `strncmp('Mock_', $className, 5)`.

Reply:
Good point -- I tend to "do what I know" rather than refresh my memory on all the string comparison functions, but this is much more elegant; I'll fix it next week.